### ISSUE TITLE
`trivially_copy_pass_by_ref` does not recognize types with `PhantomData` as trivially copyable

### ISSUE THREAD

**Issue by kjarosh** [rust-lang/rust-clippy#15813]
### Summary

When using `PhantomData`, clippy doesn't treat type as trivially copyable and `trivially_copy_pass_by_ref` ignores it.

See https://github.com/ruffle-rs/ruffle/pull/21826

### Lint Name

trivially_copy_pass_by_ref

### Reproducer

I tried this code:

```rust
#[derive(Clone, Debug, Eq, PartialEq)]
pub struct Index<T>(pub u32, pub PhantomData<T>);

impl<T: Clone> Copy for Index<T> {}

impl<T> Index<T> {
    pub fn new(i: u32) -> Index<T> {
        Index(i, PhantomData)
    }
}

fn test<T>(&mut self, i: &Index<T>) {
    // ...
}
```

I expected to see this happen:

Clippy should complain that I'm passing `Index` by reference

Instead, this happened:

Clippy did nothing :(

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04) (Fedora 1.89.0-2.fc41)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: x86_64-unknown-linux-gnu
release: 1.89.0
LLVM version: 19.1.7
```

**Comment by zihan0822** [comment]
clippy can't conclude "Index<T>" is copyable. It will lint if you further bound `T: Clone` in `test`.

**Comment by kjarosh** [comment]
I'm sorry, I don't know how we didn't notice this! Thanks for your quick response
